docs/porting/psa/TLS-on-PSA.md

```diff
@@ -0,0 +1,114 @@
+
+_Please note that this is likely to change at a later date though, as existing interfaces are likely to be deprecated in favor of the new, PSA-based APIs - if, when and how it happens is outside the scope of this document._
+
+### Switching from PSA-enabled Mbed TLS in Mbed OS 5.11 to PSA-enabled Mbed TLS in Mbed OS 5.12
+
+If your application is not using any of the new PSA-specific APIs described below, then it can be used with the version in Mbed OS 5.12 by just recompiling it. (Note that relinking is not enough, though.)
+
+If you application is using one of the new PSA-specific APIs described below, then you need to change its source code as these APIs have been change in an incompatible way between Mbed OS 5.11 and Mbed OS 5.12, to reflect changes in PSA Crypto itself, whose API wasn't stable as of Mbed OS 5.11.
+
+### New, PSA-based APIs in Mbed TLS
+
+As mentioned in the previous section, you can still build old applications with PSA-enabled Mbed TLS. Doing so is enough to benefit from PSA in some areas (for example, some PSA-based hardware acceleration depending on platform support), but in other areas changes in the application source code are necessary. This mainly means replacing calls to pre-PSA APIs with calls to their new PSA-based versions, as detailed in the following sections.
+
+The new APIs allow using PSA-managed keys for a few purposes (TLS client authentication, CSR generation, TLS PSK connections). Letting the key be managed by PSA is good for security as PSA can isolate the keys from the rest of the applications (referred to below as "opaque keys"). To use such PSA-managed keys, you'll need to modify your code to use the new APIs as described below.
+
+### Using opaque ECDSA keys for TLS client authentication
+
+This mainly involves using the new API function `mbedtls_pk_setup_opaque()` to wrap a PSA-managed key into a PK context, then using that context with the usual TLS APIs.
+
+#### Application flow without PSA
+
+1. At application startup, make sure `mbedtls_platform_setup()` is called if relevant.
+1. Declare (and allocate) an object of type `mbedtls_pk_context`.
+1. Load a key into that PK context, presumably using `mbedtls_pk_parse_key()` (or `mbedtls_pk_parse_keyfile()`).
+1. Configure the key for use in client authentication with a matching certificate by calling `mbedtls_ssl_conf_own_cert()` on the PK context.
+1. Use the resulting SSL configuration for TLS connections as usual.
+1. When you're done using TLS connections based on that configuration, _and no longer want to use that key for anything else_, free the PK context using `mbedtls_pk_free()`.
+
+#### Application flow with PSA in Mbed OS 5.11 (new/changed parts emphasized):
```

Comment:
This says we've emphasized the new/changed parts. Do we include the unchanged parts elsewhere in the Mbed OS docs, or are they just in the TLS docs? Can we link to the unchanged parts instead of including them here and bolding the new ones?